### ISSUE TITLE
Not accepting nested configurations

### ISSUE THREAD

**Issue by alexlafroscia** [jonathanKingston/broccoli-lint-eslint#26]
While developing https://github.com/jonathanKingston/ember-cli-eslint/pull/41, I initially bumped the `broccoli-lint-eslint` version to `2.0.1`, just to see if that changed anything.  I noticed that any files within the `tests/` directory of the Ember app I was testing against were no longer using the `.eslintrc` file under `tests/.eslintrc` to overwrite the configuration provided by the config file in the root of the app.  So, certain files that were passing before were failing after updating this dependency, and passed again after reverting to the original version.

**Comment by alexlafroscia** [comment]
I've added `ember-cli-eslint` to one of my Ember apps (not an addon) and am seeing this same issue.  `broccoli-lint-eslint@2.0.1` was installed by default, and the `tests/.eslintrc` config is not merged with the one in the root of the directory.  So, `ember test` fails on files in the `tests/` directory but running `eslint tests/` passes.

**Comment by BrianSipple** [comment]
I'm noticing the same thing -- will investigate further soon.

**Comment by rwjblue** [comment]
Issue is slightly but not directly related to https://github.com/jonathanKingston/broccoli-lint-eslint/blob/master/lib/index.js#L80 (which is modified by https://github.com/jonathanKingston/broccoli-lint-eslint/pull/25).

Basically, we are using `executeOnText` which uses the cached `options` (which basically includes only the root directory `.eslintrc.js` file).  Using `executeOnFile` would properly generate each files specific config.  

For this to work for us, we basically need to be able to call `executeOnText` but provide `options` (built via `CLIEngine.prototype.getConfigForFile`), but that isn't possible.  We can work around this by doing the following (I think):

```js
var config = this.cli.getConfigForFile(pathToFile);
this.cli.options = config;
this.cli.executeOnText(contents, pathToFile);
```

But it really seems odd that the ESLint side doesn't expose this without ^^ ugliness. Maybe we can ping them?

@nickiaconis - Also, RE: broccoli-persistent-filter, we would want to use `getConfigForFile` for the file specific cache key. It might be too expensive, but it would be a good place to start....

**Comment by BrianSipple** [comment]
I’ve been trying to reproduce this within the test suite for `broccoli-lint-estlint` itself, but oddly, the test helper seems to handle nested configs just fine.

I pushed up the [branch that I created](https://github.com/jonathanKingston/broccoli-lint-eslint/tree/fix-nested-configuration-errror) for starting to dig in to this if anyone else is curious, and I’d expect https://github.com/jonathanKingston/broccoli-lint-eslint/blob/fix-nested-configuration-errror/test/test.js#L22 to fail on the [nested fixture/config that I created](https://github.com/jonathanKingston/broccoli-lint-eslint/tree/fix-nested-configuration-errror/test/fixture) without yet implementing any changes to the way `this.cli` is configured and `this.cli.executeOnText` is called.

But it passes. And, when I log out the return value for `this.cli.getConfigForFile(pathToFile)` in every test, the nested fixture does, indeed, use the nested `.eslintrc`. Most likely, I’m not running things the same way that they’d be run in a consuming app that does `ember test`… I can’t quite pinpoint it yet, though.

**Comment by nickiaconis** [comment]
@rwjblue When you include a `pathToFile` in a call to `executeOnText`, ESLint [calls the equivalent of `getConfigForFile`](https://github.com/eslint/eslint/blob/87f8b18976e4c8ba0813182c64ea53f9fdfaaf10/lib/cli-engine.js#L154) under the hood.

**Comment by BrianSipple** [comment]
Should be fixed after #25
